src/Codeception/Lib/Actor/Shared/Friend.php

```diff
@@ -23,6 +23,7 @@ public function haveFriend($name, $actorClass = null)
         if (!isset($this->friends[$name])) {
             $actor = $actorClass === null ? $this : new $actorClass($this->getScenario());
             $this->friends[$name] = new LibFriend($name, $actor, $this->getScenario()->current('modules'));
+            $this->getScenario()->friends[$name] = $this->friends[$name];
```

Comment:
I'd suggest to save current friends into module and not to scenario. Scenario is a general purpose object available in all test suites, even if there are no modules with friends in it.

I'd like you to save friend instances into the WebDriver modules itself. You can get it through
```
$this->getScenario()->current('modules') // check that module implements Interfaces\MultiSession
```
and set friends into it.